File: shanny-blog-java/blog-pojo/src/main/java/com/shanny/dto/ArticleDTO.java
package com.shanny.dto;

import com.shanny.converter.LongListConverter;
import com.shanny.enums.CategoryEnum;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.util.List;

@Data
@Schema(description = "添加文章提交的数据")
public class ArticleDTO {
    private String title;
    private String content;
    private String memo;
    private String image;
    private String href;
    private List<Long> tags;
    private CategoryEnum.CategoryType type;
    private Long categoryId;
    private List<Long> timelines;
    private Integer views = 0;
    private Boolean published = true;
}

File: shanny-blog-net/blog-pojo/Dtos/ArticleDTO.cs
using blog_common.Enums;

namespace blog_pojo.Dtos
{
    public class ArticleDTO
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Content { get; set; } = string.Empty;
        public string Memo { get; set; } = string.Empty;
        public string Image { get; set; } = string.Empty;
        public string Href { get; set; } = string.Empty;
        public List<long> Tags { get; set; } = new();
        public CategoryType Type { get; set; }
        public long CategoryId { get; set; }
        public List<long> Timelines { get; set; } = new();
        public int Views { get; set; } = 0;
        public bool Published { get; set; } = true;
    }
}
